src/test/java/io/jenkins/plugins/google/analyze/code/security/accessor/IACValidationServiceTest.java

```diff
@@ -227,9 +231,9 @@ public void validateIAC_validParams_returnsNonEmptyViolations() throws IOExcepti
         when(closeableHttpClient.execute(eq(httpGet), any(ResponseHandler.class)))
                 .thenReturn(DUMMY_JSON_NON_EMPTY_VIOLATIONS_RES);
 
-        final List<Violation> violations = iacValidationService.validateIAC(getValidIACValidationRequest(listener));
+        final IaCValidationReport report = iacValidationService.validateIAC(getValidIACValidationRequest(listener));
 
-        assertEquals(violations, response.getResult().getValidationReport().getViolations());
+        assertEquals(report, response.getResult().getValidationReport());
```

Comment:
Should we not implement the equals() function for this object for deep comparison of the object?